pyccel/parser/semantic.py

```diff
@@ -356,6 +368,34 @@ def check_for_variable(self, name):
         else:
             return self.scope.find(name, 'variables')
 
+    def check_for_attribute(self, name):
+        """
+        Search for an Attribute with the given name in the current scope.
+
+        Search for an Attribute with the given name in the current scope,
+        defined by the local and global Python scopes. Return None if not found.
+
+        Parameters
+        ----------
+        name : DottedName
+            The object describing the attribute.
+
+        Returns
+        -------
+        Variable
+            Returns the attribute if found or None.
+        """
+
+        if self.current_class:
+            attributes_list = list(self._current_class.attributes)
+            for key, value in enumerate(attributes_list):
+                if value.name == name.name[-1]:
```

Comment:
Does this basic check (`value.name == name.name[-1]:`) not cause problems in the case of name collisions? Or are these not handled for classes currently anyway?
(See https://github.com/pyccel/pyccel/blob/devel/developer_docs/scope.md)

E.g:
```python
class Point:
    def __init__(self, a, A);
        self.a = a
        self.A = A
```

Reply:
> Their is no problems in the case of name collisions

Is that because name collisions aren't handled? Or because you are only testing C? Or is this handled in some unexpected way?

I would be very surprised if this works. Maybe I should give a test case which has a C name collision, rather than a fortran name collision. E.g:
```python
class Point:
    def __init__(self, a);
        self.double = a*2
```
`double` is a keyword in C so the attribute should rename itself to something like `double_0001` (I'm not sure this collision avoidance is implemented though, it is possible that names are not always verified with the Scope).
If this si the case then I think `name.name[-1]` will still be `double` but `value.name` will be `double_0001`. Is this not the case? If not then how did `name.name[-1]` become `double_0001`?

Reply:
Fixed.

Reply:
Where?

Reply:
When possible ([as mentioned in the docs](https://github.com/pyccel/pyccel/blob/devel/developer_docs/review_process.md?plain=1#L25C116-L25C192)), please leave references to commits or code when you say something is fixed so it is easy to check.

Reply:
Fixed.https://github.com/pyccel/pyccel/blob/802d795332b4c5c772830c7bd197c4b16347f873/pyccel/parser/semantic.py#L1149-L1149